.github/workflows/author_approval.yml

```
@@ -0,0 +1,43 @@
+name: Author Approval Label
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  label:
+    permissions:
+        pull-requests: write
+    runs-on: ubuntu-latest
+    # Run on comments that are:
+    # 1) on PRs, not issues,
+    # 2) not from bot users and,
+    # 3) include the string "merge approved"
+    # If so, we will do the work to check that the commenter is the package author,
+    # and conditionally apply the author-approved label.
+    # note: `merge approved` here is NOT case-sensitive, see https://docs.github.com/en/actions/learn-github-actions/expressions#contains
+    if: ${{ github.event.issue.pull_request && github.event.issue.user.type != 'Bot' && contains(github.event.comment.body, 'merge approved') }}
+    steps:
+      - name: Verify package author
+        id: verify-author
+        env:
+          # We use an env variable, not direct interpolation into the script, for security:
+          # https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable
+          PR_BODY: ${{ github.event.issue.body }}
+          COMMENTER: ${{ github.event.comment.user.login }}
+        shell: julia --compile=min --optimize=0 --color=yes {0}
+        run: |
+          m = match(r"Created by: @([^\s]+)", ENV["PR_BODY"])
```

Per the GitHub https://docs.github.com/en/enterprise-cloud@latest/admin/identity-and-access-management/iam-configuration-reference/username-considerations-for-external-authentication:

> Usernames for user accounts on GitHub.com can only contain alphanumeric characters and dashes (-).

So our regex only needs to check for alphanumeric characters (uppercase letters, lowercase letters, and numbers) and dashes.

Also, we can enforce that the "Created by" line either ends in a newline, or the end of the string.

```suggestion
          m = match(r"Created by: @([A-Za-z0-9]*+)(?:$|\n)", ENV["PR_BODY"])
```